### PR TITLE
Add health checker

### PR DESCRIPTION
**Pull request by dexchen**

### CODE REVIEW

File: cmd/health/main.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/google/uuid"
+
+	"github.com/maverics-org/maverics/pkg/endpoint/monitoring"
+	"github.com/maverics-org/maverics/pkg/service/health"
+)
+
+func main() {
+	logger := log.NewLogfmtLogger(os.Stderr)
+	logger.Log("msg", "health checker", "status", "starting")
+	portPtr := flag.Int("port", 8888, "HTTP port number")
```
Comment (eliasjf) [inline cmd/health/main.go]: Nit: let's [avoid naming variables after their types](https://dave.cheney.net/2019/01/29/you-shouldnt-name-your-variables-after-their-types-for-the-same-reason-you-wouldnt-name-your-pets-dog-or-cat)

File: pkg/endpoint/monitoring/http_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			"version": "0.0.1"}}
+	resultJSON, err := json.Marshal(result)
+	assert.NoError(t, err)
+
+	err = encodeHTTPStatusResult(nil, rw, result)
+	assert.NoError(t, err)
+
+	resp := rw.Result()
+	body, err := ioutil.ReadAll(resp.Body)
+	bodyJSON := strings.TrimRight(string(body), "\r\n")
```
Comment (eliasjf) [inline pkg/endpoint/monitoring/http_test.go]: I suggest marshaling this back to JSON as opposed to manipulating the string

File: pkg/service/health/service_test.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+func TestNew(t *testing.T) {
+	hc := New("hc-01", "health checker")
+	assert.NotNil(t, hc)
+	assert.Equal(t, "hc-01", hc.ID)
+	assert.Equal(t, "health checker", hc.Description)
+}
+
+func TestIsRunning(t *testing.T) {
```
Comment (eliasjf) [inline pkg/service/health/service_test.go]: I think we should rename this test to `TestCheckStatus`

File: pkg/service/health/service.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+	model "github.com/maverics-org/maverics/pkg/model/monitoring"
+)
+
+type HealthChecker struct {
+	ID          string
+	Description string
+}
+
+func New(id string, description string) *HealthChecker {
```
Comment (topherMarie) [inline pkg/service/health/service.go]: Idiomatically we don't need a constructor for a struct that has everything public

File: cmd/health/main.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	}
+	m.Handle("/status", healthendpoint.NewHTTPHandler(e, options...))
+
+	httpListener, err := net.Listen("tcp", ":"+strconv.Itoa(*port))
+	if err != nil {
+		level.Info(logger).Log("error", "failed to create a listener "+err.Error())
+		os.Exit(1)
+	}
+
+	level.Info(logger).Log("info", "health checker listens on :"+strconv.Itoa(*port))
```
Comment (eliasjf) [inline cmd/health/main.go]: This log entry will already contain `level=info`, so it doesn't make much sense to include "info" again. I'd recommend `level.Info(logger).Log("msg", "health checker listens on :"+strconv.Itoa(*port))`

File: cmd/health/main.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+	level.Info(logger).Log("error", http.Serve(httpListener, m))
```
Comment (eliasjf) [inline cmd/health/main.go]: For structural consistency, what do you think of `level.Info(logger).Log("msg", http.Serve(httpListener, m))`?

Comment (dexchen) [reply cmd/health/main.go]: It is a good question. The question is whether we need to differentiate error messages from informational messages. Or shall we use "msg" as the key for all?

File: pkg/service/health/service.go (hunk 4 above)
Comment (topherMarie) [reply pkg/service/health/service.go]: I'm not sure why we'd make the fields private? Better to remove the constructor and tests.

**Pull request merged**